packages/smooth_app/lib/helpers/picture_capture_helper.dart

```diff
@@ -1,39 +1,105 @@
+import 'dart:io';
+import 'dart:math';
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
 import 'package:openfoodfacts/openfoodfacts.dart';
 import 'package:provider/provider.dart';
 import 'package:smooth_app/data_models/continuous_scan_model.dart';
 import 'package:smooth_app/database/product_query.dart';
-import 'package:smooth_app/generic_lib/loading_dialog.dart';
+import 'package:workmanager/workmanager.dart';
+
+@pragma(
+    'vm:entry-point') // Mandatory if the App is obfuscated or using Flutter 3.1+
+void callbackDispatcher() {
+  Workmanager().executeTask(
+    (String task, Map<String, dynamic>? inputData) async {
+      // make a counter with task as key as it is unique for each task
+      final int counter = inputData!['counter'] as int;
+      // if task is greate than 6 , that means it has been executed 7 times
+      if (counter > 6) {
+        // returns true to let platform know that the task is completed
+        final File file = File(inputData['imageUri'].toString());
+        file.delete();
+        return Future<bool>.value(true);
+      }
+      final List<Duration> duration = <Duration>[
+        const Duration(seconds: 30),
+        const Duration(minutes: 1),
+        const Duration(minutes: 30),
+        const Duration(hours: 1),
+        const Duration(hours: 6),
+        const Duration(days: 1),
+      ];
+      bool shouldRetry = false;
+      try {
+        final SendImage image = SendImage(
```

Comment:
Before starting the upload, you should check a network is available (or at least say it is not in airplane mode)

Reply:
```dart
 await Workmanager().registerOneOffTask(
          task,
          'ImageUploadWorker',
          constraints: Constraints(
            networkType: NetworkType.connected,
          ),
          inputData: inputData,
          initialDelay: duration[counter],
        );
```
 I belive this constraint is taking care of this

Reply:
Yes I know, but the extra check would be better.
Internet access != OFF server online (= basically try to ping the server)

Reply:
```dart
import 'dart:io';
  

Future<void> main() async {
  final HttpClient client = HttpClient();
  try{
    final HttpClientRequest request = await client.get(
    'world.openfoodfacts.org' , 80 , '/'
    );
    
    final HttpClientResponse response = await request.close();
    print(response.statusCode);
  }
  catch(e){
    print(e);
  }
}
```
ok something like this ? but when it doesn't return 200 then what the retry method here shoud be  kinda confused ??? , like i should check ping again but after how much time , and won't it be bad for end user like even if it retries after every 30 seconds to check for ping again the work thread is still doing it's work ?? using sleep of 30s then sounds good ?
Suggestions ?? @g123k

Reply:
I am getting a bit confusing here what actually u are trying to say to me(not getting enough context),
- Like if ping fails i still do the schedule of task for duration[tries] time ?
- or i just sleep for duration[tries] time 
If i go with the first one then it's not needed cuz it's already being handled ,scheduled if it fails to update anyhow
if i go with the later on, won't it be that a task is there running which should rather be scheduled